internal/querynodev2/delegator/delete_buffer.go

```diff
@@ -0,0 +1,53 @@
+// Licensed to the LF AI & Data foundation under one
+// or more contributor license agreements. See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership. The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package delegator
+
+import (
+	"github.com/milvus-io/milvus/internal/storage"
+	"github.com/milvus-io/milvus/internal/util/typeutil"
+)
+
+type deleteBufferItem struct {
+	partitionID int64
+	deleteData  storage.DeleteData
+}
+
+// deleteBuffer caches L0 delete buffer for remote segments.
+type deleteBuffer struct {
+	// timestamp => DeleteData
+	cache *typeutil.SkipList[uint64, []deleteBufferItem]
```

Comment:
I chose to use SkipList here because:
- We need to truncate delete data before a safe ts. Using a slice might cause copying remaining data or left truncated data in it(cannot be GCed). So it needs to be linkedlist-ish
- Need to locate data after ts quick, so linked-list will have O(n) complexity to search